hackathons.md

```diff
@@ -0,0 +1,563 @@
+
+- Uniquely identifying rows with a `PRIMARY KEY`
+- Using columns to relate tables together with `FOREIGN KEY`s
+
+#### Primary Key
+
+{{site.data.alerts.callout_success}}<strong>tl;dr</strong>: You need a column to uniquely identify each row, called a <code>PRIMARY KEY</code>.{{site.data.alerts.end}}
+
+With a database, it's important to plan to create a column (or columns) that can uniquely identify each row––in relational databases (including CockroachDB) this is known as a `PRIMARY KEY`. This column:
+
+- Must contain unique values for each row
+- Cannot be `null`
+
+One "gotcha" with CockroachDB is that you currently cannot add a `PRIMARY KEY` after creating a table. For hackathon apps, it's really not a big deal, but it's good practice to be mindful of this decision.
+
+#### Foreign Key
+
+{{site.data.alerts.callout_success}}<strong>tl;dr</strong>: To relate two tables together use a <code>FOREIGN KEY</code>, which represents a column from one table in another.{{site.data.alerts.end}}
+
+Though this typically works itself out with a solid object-oriented design, it's important to understand tables that you want to relate with one another and ensure that you include columns that will let you do that.
+
+For instance, if you have tables for `customers` and the `orders` they place, you would want to include a column in the `orders` table that lets you identify which row from the `customers` table it relates to; typically the `PRIMARY KEY` of the referenced table. This is often called a `FOREIGN KEY`.
+
+By creating this relationship you can easily do things like find all of a user's orders, or do more complicated things like filter `order` data based on data that exists only in the `customers` table with a `JOIN` (more on that later).
+
+SQL offers an explicit `FOREIGN KEY` constraint, which can ensure your data remains consistent. However, it's also possible to 
+
+## 3. Implement Your Schema
+
+So, it's important to understand what kind of data you want to store––i.e., your schema––and then implement it in your database.
+
+Here's how to do that:
+
+1. Create a database:
+
+    ~~~ sql
+    > CREATE DATABASE db_name;
+    ~~~
+
+    If this is the only database you're using, you can set it as the default database:
+
+    ~~~ sql
+    > SET DATABASE = db_name;
+    ~~~
+
+2. Create tables for your data:
+
+    ~~~ sql
+    > CREATE TABLE tbl_name (
+      col1 TYPE PRIMARY KEY,
+      col2 TYPE,
+      etc.
+    );
+    ~~~
+
+    Note that we denote which column we want to use to uniquely identify the column by identifying it as the `PRIMARY KEY`.
+
+    You'll need to create a table for each type of data (or "class") you want to store.
+
+### Implementing `FOREIGN KEY` (Optional)
+
+ For some of your tables, you'll want to include a `FOREIGN KEY` columns. You don't have to formally enforce the `FOREIGN KEY` relationship, but you can to ensure you don't end up with inconsistent data:
+
+ ~~~ sql
```

Comment:
Need to add prompts before each statement. Might be nice to split them into separate code blocks and add "copy-to-clipboard" buttons.